sympy/integrals/integrals.py

```diff
@@ -516,6 +517,38 @@ def try_meijerg(function, xab):
                             function = ret
                             continue
 
+            if not isinstance(antideriv,Integral) and antideriv is not None:
+                sym = xab[0]
+                if antideriv.is_Add:
+                    terms = antideriv.args
+                else:
+                    terms = tuple((antideriv,))
+                for term in terms:
+                    atan_part = term.atoms(atan)
+                    if atan_part and len(atan_part)==1:
+                        atan_arg = atan_part.pop().args[0]
```

Comment:
I am not sure if it is necessary to check `len(atan_part) == 1`. I would consider something like

    for atan_term in atan_part:
        atan_arg = atan_term.args[0]
        for tan_term in atan_arg.atoms(tan):